src/main/java/org/openpnp/machine/reference/feeder/wizards/ReferenceTubeFeederConfigurationWizard.java

```diff
@@ -19,13 +19,13 @@
 
 package org.openpnp.machine.reference.feeder.wizards;
 
-import org.openpnp.machine.reference.feeder.ReferenceTubeFeeder;
+import org.openpnp.machine.reference.ReferenceFeeder;
 
 public class ReferenceTubeFeederConfigurationWizard
         extends AbstractReferenceFeederConfigurationWizard {
-    private final ReferenceTubeFeeder feeder;
+    private final ReferenceFeeder feeder;
```

Comment:
Adaptive threshold is only used sometimes for downlooking camera,
the usage is really limited and there are already many stages. I remove it.
If you want, i can update convertColor, but with FluentCv removed, i
post the code at the
end of this mail.

I have tried to use the abstract wizard, but got this compiling error:

[ERROR] Failed to execute goal
org.apache.maven.plugins:maven-compiler-plugin:3.1:compile
(default-compile) on project openpnp-gui: Compilation failure
[ERROR] /home/av/test/work/openpnp/openpnp-develop/x/openpnp-develop/new/openpnp-develop/src/main/java/org/openpnp/machine/reference/feeder/Bank.java:[223,17]
org.openpnp.machine.reference.feeder.wizards.AbstractReferenceFeederConfigurationWizard
is abstract; cannot be instantiated



----

package org.openpnp.vision.pipeline.stages;

import java.util.ArrayList;
import org.opencv.core.Mat;
import org.opencv.core.Core;
import org.opencv.imgproc.Imgproc;
import org.openpnp.vision.pipeline.CvPipeline;
import org.openpnp.vision.pipeline.CvStage;
import org.simpleframework.xml.Attribute;

/**
 * Converts the color of the current working image to the specified conversion.
 */
public class ConvertColor extends CvStage {


// copyed from fluentCV
    public enum ColorCode {
        Bgr2Gray(Imgproc.COLOR_BGR2GRAY),
        Bgr2Lum(-Imgproc.COLOR_BGR2Luv),
        Bgr2YCrCb(Imgproc.COLOR_BGR2YCrCb),
        YCrCb2Bgr(Imgproc.COLOR_YCrCb2BGR),
        Gray2Bgr(Imgproc.COLOR_GRAY2BGR),
        Bgr2Hls(Imgproc.COLOR_BGR2HLS),
        Hls2Bgr(Imgproc.COLOR_HLS2BGR),
        Bgr2HlsFull(Imgproc.COLOR_BGR2HLS_FULL),
        Hls2BgrFull(Imgproc.COLOR_HLS2BGR_FULL),
        Bgr2Hsv(Imgproc.COLOR_BGR2HSV),
        Hsv2Bgr(Imgproc.COLOR_HSV2BGR),
        Bgr2HsvFull(Imgproc.COLOR_BGR2HSV_FULL),
        Hsv2BgrFull(Imgproc.COLOR_HSV2BGR_FULL);

        private int code;

        ColorCode(int code) {
            this.code = code;
        }

        public int getCode() {
            return code;
        }
    }

    @Attribute
    private ColorCode conversion = ColorCode.Bgr2Gray;

    public ColorCode getConversion() {
        return conversion;
    }

    public void setConversion(ColorCode conversion) {
        this.conversion = conversion;
    }

    @Override
    public Result process(CvPipeline pipeline) throws Exception {
        Mat mat = pipeline.getWorkingImage();
	if(conversion.getCode()<0) {
		ArrayList<Mat> channels = new ArrayList<Mat>(3);
        	Imgproc.cvtColor(mat, mat, -conversion.getCode());
    		Core.split(mat, channels);
		mat.release(); mat=null;
		mat = channels.get(0);
	} else {
        	Imgproc.cvtColor(mat, mat, conversion.getCode());
	}
        return new Result(mat);
    }
}


2016-07-26 17:32 GMT, Cri S <phone.cri@gmail.com>:
> As TubeFeeder actually is the feeder with less features, basic feeder
> wizard, it's used
> for other feeders, this wizard could be used by other feeders that
> need just minimal configurations.
> Should i copy this as ReferenceFeederConfigurationWizard.java ? is this
> better ?
>
> 2016-07-26 17:28 GMT, Jason von Nieda <notifications@github.com>:
>>>
>>>  public class ReferenceTubeFeederConfigurationWizard
>>>          extends AbstractReferenceFeederConfigurationWizard {
>>> -    private final ReferenceTubeFeeder feeder;
>>> +    private final ReferenceFeeder feeder;
>>
>> What is the purpose of this change? It limits access to any feeder
>> specific
>> properties and this is not how any of the other wizards work.
>>
>> ---
>> You are receiving this because you authored the thread.
>> Reply to this email directly or view it on GitHub:
>> https://github.com/openpnp/openpnp/pull/305/files/15ab3f127e639a1e905fb7668367edd1280810a3#r72297218
>